server/chat-plugins/abuse-monitor.ts

```diff
@@ -0,0 +1,663 @@
+/**
+ * Neural net chat'filters'.
+ * These are in a separate file so that they don't crash the other filters.
+ * (issues with globals, etc)
+ * We use Google's Perspective API to classify messages.
+ * @see https://perspectiveapi.com/
+ * by Mia.
+ * @author mia-pi-git
+ */
+
+import {FS, Utils, Net, ProcessManager, Repl} from '../../lib';
+import {Config} from '../config-loader';
+import {toID} from '../../sim/dex-data';
+import {throttling} from '../../tools/set-import/importer';
```

Comment:
ugh pls no

Reply:
Please don't change `set-import` or introduce new dependencies in crazy ways. `tools` should be self-contained.

Reply:
The throttle in there can't be audited, anyway. Easy enough to just reimplement; maybe put an implementation in `Utils`.

Reply:
Okay, done.

Reply:
Moved it to utils and restored set-import.